### PR TITLE
Various fixes (again⁴)

### PR DESCRIPTION
**Pull request by kit-ty-kate**
This PR should fix all the broken packages available in OCaml 4.08 and 4.09. The only remaining ones are broken because of C libraries.

Work done using [check.ocamllabs.io](http://check.ocamllabs.io)

A few of those broken packages need future attentions from their maintainers as it involves the latest version of their packages:
* `why3` is not compatible with OCaml >= 4.09 (cc @silene)
* `utop` is not compatible with OCaml >= 4.09 (cc @diml)
* `spoc` is not compatible with OCaml >= 4.08 (cc @mathiasbourgoin)
* `snappy` is not compatible with OCaml >= 4.09 (cc @ygrek)
* `relit-reason` needs to fix its opam file upstream (cc @charlesetc)
* `regenerate` needs to fix its opam file upstream (cc @Drup)
* `reason-parser` needs to fix its opam file upstream (cc @jordwalke)
* `pxp` is not compatible with OCaml >= 4.08 (cc @codinuum)
* `archimedes` needs to fix its opam file upstream (cc @Chris00)
* `imagemagick` is not compatible with safe-strings (cc @vouillon)
* `earlybird` is not compatible with OCaml >= 4.08 (cc @hackwaly)
* `flow_parser` is not compatible with OCaml >= 4.08 (cc @gabelevi)
* `electrod` needs to fix its opam file upstream (cc @grayswandyr)
* `mlgmpidl` is not compatible with OCaml >= 4.09 (cc @nberth)
* `deriving-yojson` is not compatible with camlp4 >= 4.08 (cc @hhugo) (cc @diml it looks like a camlp4 bug, same with `sqlgg.0.4.3`)
* `malfunction` is not compatible with OCaml >= 4.09 (cc @stedolan)
* `odoc` is not compatible with OCaml >= 4.09 (cc @aantron)
* `ivy` is not compatible with safe-strings (cc @flixr)
* `fstar` is not compatible with OCaml >= 4.08 (cc @tahina-pro)
* `milter` is not compatible with OCaml >= 4.09 (cc @andrenth)
* `eliom` is not compatible with OCaml >= 4.08 (cc @jrochel)
* `asak` is not compatible with OCaml >= 4.09 (cc @nobrakal)
* `augeas` is not compatible with OCaml >= 4.09 (cc @Leonidas-from-XIV)
* `bamboo` is not compatible with OCaml >= 4.08 (cc @pirapira)
* `bogue` needs to fix its opam file upstream (cc @sanette)
* `brotli` is not compatible with safe-strings and needs to fix its opam file upstream (cc @fxfactorial)
* `caml2html` is not compatible with OCaml >= 4.03 (cc @mjambon)
* `camllib` is not compatible with OCaml >= 4.09 (cc @nberth)
* `gperftools` is not compatible with OCaml >= 4.09 (cc @ygrek)
* `dokeysto_lz4` needs to fix its opam file upstream (cc @UnixJunkie)
* `dolmen` is not compatible with dune >= 1.10 (cc @Gbury)
* `io` needs to fix its opam file upstream (cc @rizo)
* `lablqml` is not compatible with OCaml >= 4.09 (cc @Kakadu)
* `leveldb` is not compatible with OCaml >= 4.09 (cc @mfp)
* `numerix` is not compatible with safe-strings (cc @iksnalybok)
* `ocamlviz` is not compatible with safe-strings (cc @lefessan)
* `ocsfml` needs to fix its opam file upstream (cc @kyoDralliam)
* `ocsigen-i18n` is not compatible with OCaml >= 4.08 (cc @jrochel)
* `ocaml-protoc-yojson` needs to fix its opam file upstream (cc @mransan)

### CODE REVIEW

**Comment by mseri** [comment]
- `brotli` and `pulseaudio` failures are expected because they require a missing c dependency

- `humane-re` seems to require `unsafe-string`

- `ocsfml` is failing to compile its C++ part

- I am not sure about `owl-zoo`, that version predates my times, but I think it needs at least a bound on `owl` and `owl-base` < 0.4.0.

- Finally a source file has changed: ```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - statverif.1.97pl1:
            https://github.com/rittere/statverif/archive/1.97pl1.tar.gz (Bad checksum, expected md5=6e4ab2c56c4af4cb001c8e6353ad00d5)
```

File: packages/owl-zoo/owl-zoo.0.3.7/opam [hunk 1]
```diff
@@ -16,7 +16,7 @@ depends: [
   "ocaml" {>= "4.06.0"}
   "jbuilder" {build}
   "ocaml-compiler-libs"
-  "owl" {< "0.5.0"}
+  "owl" {< "0.4.0"}
```
Comment (mseri) [inline packages/owl-zoo/owl-zoo.0.3.7/opam]:
```suggestion
  "owl" {< "0.4.0"}
  "owl-base" {< "0.4.0"}
```

File: packages/humane-re/humane-re.0.0.2/opam [hunk 2]
```diff
@@ -18,6 +18,7 @@ depends: [
   "oasis"
   "re"
   "ocamlbuild" {build}
+  "ounit"
```
Comment (mseri) [inline packages/humane-re/humane-re.0.0.2/opam]:
```suggestion
  "ounit"
   "base-unsafe-string"
```

File: packages/humane-re/humane-re.0.0.1/opam [hunk 3]
```diff
@@ -17,6 +17,7 @@ depends: [
   "oasis"
   "re"
   "ocamlbuild" {build}
+  "ounit"
```
Comment (mseri) [inline packages/humane-re/humane-re.0.0.1/opam]:
```suggestion
  "ounit"
   "base-unsafe-string"
```

**Comment by UnixJunkie** [comment]
Thanks for the fix for package dokeysto_lz4 (v3.0.0).
However, the latest version (v3.0.1) doesn't have this problem, so the fix is just needed in the opam-repository.

**Pull request merged**